src/partialHydration/mountComponentsInHtml.ts

```diff
@@ -14,7 +14,7 @@ export default function mountComponentsInHtml({ page, html, hydrateOptions }): s
   let outputHtml = html;
   // sometimes svelte adds a class to our inlining.
   const matches = outputHtml.matchAll(
-    /<div class="ejs-component[^]*?" data-ejs-component="([A-Za-z]+)" data-ejs-props="({[^]*?})" data-ejs-options="({[^]*?})"><\/div>/gim,
+    /<.+? class="ejs-component[^]*?" data-ejs-component="([A-Za-z]+)" data-ejs-props="({[^]*?})" data-ejs-options="({[^]*?})"><\/.+?>/gim,
```

Comment:
Only thing I'm unsure of is if there is a way we can ensure that these two tags match. My regex isn't amazing, but I'm always concerned about greedy calls.

Reply:
Should be possible with a backreference. I also replaced `.+?` with `\S+` to have a narrower match.

```suggestion
    /<(\S+) class="ejs-component[^]*?" data-ejs-component="([A-Za-z]+)" data-ejs-props="({[^]*?})" data-ejs-options="({[^]*?})"><\/\1>/gim,
```

Reply:
Thanks @vollnhals, I have implement this in https://github.com/Elderjs/elderjs/pull/221/commits/e74afb79c1e20d4b99cab6d181345b1927118226